DaddyWarbucks/feathers-google-maps
# refactor: typescript

This PR is somewhat opinionated. Besides the typescript refactor, I made a few changes:

- use vite/vitest for testing
- use [unbuild](https://www.npmjs.com/package/unbuild) for building cjs & esm
- use npm instead of yarn
- use eslintrc with prettier
- rename 'lib/' to 'src/'
- use [np](https://www.npmjs.com/package/np) for publishing (just run `npm run release`)

Please let me know if you dislike one or more of these changes and I'm happy to revert.

See the typescript benefits:
<img width="581" alt="Bildschirm­foto 2023-04-02 um 20 42 37" src="https://user-images.githubusercontent.com/22286818/229373590-d41cac13-e9e3-4536-8f33-3eef8c8090b7.png">

<img width="370" alt="Bildschirm­foto 2023-04-02 um 20 43 16" src="https://user-images.githubusercontent.com/22286818/229373596-560cc9b6-1286-471f-b5e0-a7fa3b39c4eb.png">
